### PR TITLE
Improve MatrixObj documentation

### PR DESCRIPTION
**Pull request by ssiccha**
Make section on required operations for new MatrixObj objects more
discoverable: rename it and add a reference to the introduction
of its chapter.

### CODE REVIEW

**Review by ThomasBreuer — COMMENTED** [review]
I do not understand the change of the section heading. Also the other operations are "required", the point is that one has to provide specific methods for the ones without default methods when one implements a new kind of vector or matrix objects. The proposed heading does not express this.

**Comment by ssiccha** [comment]
I agree that "required operations" does not fit here. I don't like the current name either though.

IMO, the idea behind this chapter is to list the minimal set of methods one needs to install for a new type of vector or matrix objects to make it work. So I'll change the name of the chapter to "Implementing New Vector and Matrix Objects Types". What do you think?

**Comment by ThomasBreuer** [comment]
O.K.
(It is not the title of the chapter but the title of the section.)

File: lib/matobj2.gd [hunk 1]
```diff
@@ -130,6 +130,10 @@
 ##  However, row list matrices behave nearly like lists of row vectors
 ##  that insist on being dense and containing only vectors of the same
 ##  length and with the same base domain.
+##  <P/>
+##  New <Ref Filt="IsMatrixObj"/> and <Ref Filt="IsVectorObj"/> representations
+##  have to implement methods for the operations listed in Section
+##  <Ref Sect="Implementing New Vector and Matrix Objects Types"/>
```
Comment (fingolfin) [inline lib/matobj2.gd:136]:
```suggestion
##  <Ref Sect="Implementing New Vector and Matrix Objects Types"/>.
```

Comment (fingolfin) [reply lib/matobj2.gd:136]: Hmm, but of course this is just the minimum; in practice some more will almost always be implemented for performance? But anyway, shouldn't this be explained in detail in section "Implementing New Vector and Matrix Objects Types"; and then we just put a link to that here?

Comment (ssiccha) [reply lib/matobj2.gd:136]: Jepp, I've replaced this by only a link to that section.

**Comment by ssiccha** [comment]
Rebased onto master.

File: lib/matobj2.gd (hunk 1 above)
Comment (fingolfin) [reply lib/matobj2.gd:153]: Will you still apply my fix (adding a missing `.`)?

Comment (ssiccha) [reply lib/matobj2.gd]: Done.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**